extensions/hawkbit-extension-artifact-repository-s3/src/main/java/org/eclipse/hawkbit/artifact/repository/S3Artifact.java

```diff
@@ -10,28 +10,35 @@
 
 import java.io.InputStream;
 
-import org.eclipse.hawkbit.artifact.repository.model.DbArtifact;
+import org.eclipse.hawkbit.artifact.repository.model.AbstractDbArtifact;
+import org.eclipse.hawkbit.artifact.repository.model.DbArtifactHash;
 
 import com.amazonaws.services.s3.AmazonS3;
 
 /**
- * An {@link DbArtifact} implementation which retrieves the {@link InputStream}
+ * An {@link AbstractDbArtifact} implementation which retrieves the {@link InputStream}
  * from the {@link AmazonS3} client.
  */
-public class S3Artifact extends DbArtifact {
+public class S3Artifact extends AbstractDbArtifact {
 
     private final AmazonS3 amazonS3;
     private final S3RepositoryProperties s3Properties;
     private final String key;
 
-    S3Artifact(final AmazonS3 amazonS3, final S3RepositoryProperties s3Properties, final String key) {
+    S3Artifact(final AmazonS3 amazonS3, final S3RepositoryProperties s3Properties, final String key,
+            final String artifactId, final DbArtifactHash hashes, final Long size, final String contentType) {
+        super(artifactId, hashes, size, contentType);
         this.amazonS3 = amazonS3;
         this.s3Properties = s3Properties;
         this.key = key;
     }
 
     @Override
     public InputStream getFileInputStream() {
+        if (amazonS3 == null) {
```

Comment:
like in the `ArtifactFilesystem` implementation I wouldn't return `null` in the `getFileInputStream` because a caller would not check `null` when calling `getFileInputStream` better throw a runtimeexception in case of failure and exception text tells you what's wrong.